tasks/windows.yml

```diff
@@ -13,36 +13,96 @@
-  register: result
-  retries: 5
-  delay: 10
-  until: result.rc == 0
-  check_mode: false
-  changed_when: "'No changes made.' not in result.stdout_lines"
-  notify: "restart ops-agent agent windows"
+    - name: Add repo and install agent or remove repo and uninstall agent
+      win_shell: >
+        .\\add-google-cloud-ops-agent-repo.ps1 {{ '-AlsoInstall' if package_state == 'present' else '-Uninstall -RemoveRepo' }} -Version {{ version }}
+        {{ '-WhatIf' if ansible_check_mode else ''}}
+      args:
+        chdir: "{{ tempfolder.path }}"
+      register: result
+      retries: 5
+      delay: 10
+      until: result.rc == 0
+      check_mode: false
+      changed_when: "'No changes made.' not in result.stdout_lines"
+      notify: "restart windows {{ agent_type }} agent"
+
+    - when: package_state == 'present'
+      block:
+        - name: Copy main config file onto the remote machine
+          win_copy:
+            src: "{{ main_config_file }}"
+            dest: 'C:\Program Files\Google\Cloud Operations\Ops Agent\config\config.yaml'
+            force: true
+          when: main_config_file | length > 0
+          notify: "restart windows {{ agent_type }} agent"
 
-- when: package_state == 'present'
+- when: agent_type == 'logging'
   block:
-    - name: Copy main config file onto the remote machine
-      win_copy:
-        src: "{{ main_config_file }}"
-        dest: 'C:\Program Files\Google\Cloud Operations\Ops Agent\config\config.yaml'
-        force: true
-      when: main_config_file | length > 0
-      notify: "restart ops-agent agent windows"
+    - name: Validate version
+      assert:
+        that: version == 'latest'
+        msg: "Received invalid version: '{{ version }}'. The Cloud Ops Ansible role supports the following version for the Windows logging agent: 'latest'."
+
+    - when: package_state == 'present'
+      block:
+        - name: Download executable
+          win_get_url:
+            url: "https://dl.google.com/cloudagents/windows/StackdriverLogging-v1-14.exe"
+            dest: "{{ tempfolder.path }}/StackdriverLogging-v1-14.exe"
+          check_mode: false
+          changed_when: false
+
+        - name: Install agent
+          win_package:
+            product_id: GoogleStackdriverLoggingAgent
+            path: "{{ tempfolder.path }}/StackdriverLogging-v1-14.exe"
+            arguments: /S
+          notify: "restart windows {{ agent_type }} agent"
+
+        - name: Copy main config file onto the remote machine
+          win_copy:
+            src: "{{ main_config_file }}"
+            dest: 'C:\Program Files (x86)\Stackdriver\LoggingAgent\fluent.conf'
+            force: true
+          when: main_config_file | length > 0
+          notify: "restart windows {{ agent_type }} agent"
+
+        - name: Copy additional configs onto the remote machine
+          win_copy:
+            src: "{{ item }}"
+            dest: 'C:\Program Files (x86)\Stackdriver\LoggingAgent\config.d\'
```

Comment:
We might need another step to enable customized `config.d` directory. Probably needs to set an environment variable `CUSTOM_CONFIG_PLACE_HOLDER` to the `C:\Program Files (x86)\Stackdriver\LoggingAgent\config.d\` directory.

Some reference: https://github.com/GoogleCloudPlatform/google-fluentd/blob/master/windows-installer/fluent-template.conf#L3

Reply:
It looks like this gets set to "config.d/**/*.conf' by default. 
https://github.com/GoogleCloudPlatform/google-fluentd/blob/442b371bc49cbf6046a1c0a38c98b62d8fad926c/windows-installer/setup.nsi#L230

I suppose there could be an issue if the user specifies a different CUSTOM_CONFIG_PLACE_HOLDER with their own custom config directory. Then we'd be placing the plugins under config.d instead of the directory the specified. 

I think the two options are:
1. Parse fluentd.conf to get the custom config dir from the include statement.
2. If it's okay to have multiple includes we could just include a step that verifies that "@include config.d/**/*.conf" is present in fluent.conf.

Reply:
>>> It looks like this gets set to "config.d/**/*.conf' by default.

Ah I see. Then this is good enough. If the user uses Ansible to manage the custom configuration, we can make the directory path on the VM fixed to `C:\Program Files (x86)\Stackdriver\LoggingAgent\config.d\` for now.